### PR TITLE
DOC Document jQuery changes

### PR DESCRIPTION
**Pull request by GuySartorelli**
Documents the jQuery upgrade

## Parent issue
- https://github.com/silverstripe/silverstripe-admin/issues/1351

### CODE REVIEW

**Comment by PRO-BROS** [comment]
Very well. Facing same issue but now resolved on my [site](https://prhocoding.com/)

File: en/04_Changelogs/4.12.0.md [hunk 1]
```diff
@@ -23,6 +24,40 @@ While it is still advised that you perform your own due diligence when upgrading
...
+
+The short answer is that your CMS javascript customisations are unlikely to be affected unless your code is very old (e.g. you have a site upgraded from CMS 3 which has CMS javascript customisations from that time).
+
+##### jQuery API changes {#jquery-api-changes}
+
+Very little has actually changed with the jQuery API itself over this time, for the most part any removed API was removed years ago. It's likely that most projects haven't been using old jQuery API unless they've inherited old code from CMS 3 customisations.
+
+jQuery provide [a list of deprecated API](https://api.jquery.com/category/deprecated/) which also marks any API which has been removed. Most of these have recommendations for replacing them as well, so updating your code should be pretty straight forward.
+
+While it is best practice to stop using any deprecated API, you should be able to just do a quick search through your backend javascript and replace anything which the list marks as _removed_ from jQuery - everything else was deprecated but is still present and working for the time being.
```
Comment (michalkleiner) [inline en/04_Changelogs/4.12.0.md]: for consistency...

```suggestion
While it is best practice to stop using any deprecated API, you should be able to just do a quick search through your back-end javascript and replace anything which the list marks as _removed_ from jQuery - everything else was deprecated but is still present and working for the time being.
```

**Review by michalkleiner — APPROVED** [review]

File: en/04_Changelogs/4.12.0.md (hunk 1 above)
Comment (ChloeSartorelli) [reply en/04_Changelogs/4.12.0.md]: Done

**Comment by michalkleiner** [comment]
@GuySartorelli not sure if this can be merged on its own before the other PRs are merged so leaving this with you.

**Comment by GuySartorelli** [comment]
Yeah I'd hold off merging until the actual change is merged. It's in peer review, so whoever reviews the main work will merge this as well. 👍

File: en/04_Changelogs/4.12.0.md [hunk 2]
```diff
@@ -23,6 +24,40 @@ While it is still advised that you perform your own due diligence when upgrading
 
 ## Features and enhancements {#features-and-enhancements}
 
+### jQuery updated from 1.7 to 3.6 {#jquery}
+
+We've upgrade jQuery in the CMS from 1.7 to 3.6. Note that this won't affect your front-end at all unless you've been using back-end javascript in your front-end, which is heavily discouraged.
```
Comment (emteknetnz) [inline en/04_Changelogs/4.12.0.md]:
```suggestion
We've upgraded jQuery in the CMS from 1.7 to 3.6. Note that this won't affect your front-end at all unless you've been using back-end javascript in your front-end, which is heavily discouraged.
```

File: en/04_Changelogs/4.12.0.md [hunk 3]
```diff
@@ -23,6 +24,40 @@ While it is still advised that you perform your own due diligence when upgrading
...
+
+#### Will this break my CMS customisations? {#jquery-what-will-break}
+
+The short answer is that your CMS javascript customisations are unlikely to be affected unless your code is very old (e.g. you have a site upgraded from CMS 3 which has CMS javascript customisations from that time).
+
+##### jQuery API changes {#jquery-api-changes}
+
+Very little has actually changed with the jQuery API itself over this time, for the most part any removed API was removed years ago. It's likely that most projects haven't been using old jQuery API unless they've inherited old code from CMS 3 customisations.
+
+jQuery provide [a list of deprecated API](https://api.jquery.com/category/deprecated/) which also marks any API which has been removed. Most of these have recommendations for replacing them as well, so updating your code should be pretty straight forward.
```
Comment (emteknetnz) [inline en/04_Changelogs/4.12.0.md]:
```suggestion
jQuery provides [a list of deprecated API](https://api.jquery.com/category/deprecated/) which also marks any API which has been removed. Most of these have recommendations for replacing them as well, so updating your code should be pretty straight forward.
```

File: en/04_Changelogs/4.12.0.md [hunk 4]
```diff
@@ -23,6 +24,40 @@ While it is still advised that you perform your own due diligence when upgrading
...
+
+If you're using any jQuery extensions or plugins, it will pay to make sure these are up to date.
+
+##### jQuery behaviour changes {#jquery-behaviour-changes}
+
+jQuery has made some breaking behavioural changes over time. If you are experiencing problems after updating your usage of the jQuery API and any plugins you're using, you might want to check the [jQuery upgrade guides](https://jquery.com/upgrade-guide/) to see if there is a solution to your problem there.
+
+##### Entwine behavioural change {#jquery-entwine-change}
+
+The only change to behaviour we've had to make to accomodate jQuery's breaking changes was with entwine's `onadd` and `onmatch` handlers. jQuery has made substantial changes to its internal functionality, especially in regards to how it adds elements to the DOM, so we had to change with it. Entwine used to change jQuery's internal functionality to fire events synchronously when adding elements to the DOM. It now uses a `MutationObserver` to detect elements being added to the DOM, which means the events are fired asyncronously.
```
Comment (emteknetnz) [inline en/04_Changelogs/4.12.0.md]:
```suggestion
The only change to behaviour we've had to make to accommodate jQuery's breaking changes was with entwine's `onadd` and `onmatch` handlers. jQuery has made substantial changes to its internal functionality, especially in regards to how it adds elements to the DOM, so we had to change with it. Entwine used to change jQuery's internal functionality to fire events synchronously when adding elements to the DOM. It now uses a `MutationObserver` to detect elements being added to the DOM, which means the events are fired asynchronously.
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: en/04_Changelogs/4.12.0.md (hunk 4 above)
Comment (ChloeSartorelli) [reply en/04_Changelogs/4.12.0.md]: Done

File: en/04_Changelogs/4.12.0.md (hunk 3 above)
Comment (ChloeSartorelli) [reply en/04_Changelogs/4.12.0.md]: Done

File: en/04_Changelogs/4.12.0.md (hunk 2 above)
Comment (ChloeSartorelli) [reply en/04_Changelogs/4.12.0.md]: Done

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**